starlark/eval.go

```diff
@@ -836,14 +826,50 @@ unknown:
 	return nil, fmt.Errorf("unknown binary op: %s %s %s", x.Type(), op, y.Type())
 }
 
-func repeat(elems []Value, n int) (res []Value) {
-	if n > 0 {
-		res = make([]Value, 0, len(elems)*n)
-		for i := 0; i < n; i++ {
-			res = append(res, elems...)
-		}
+// It's always possible to overeat in small bites but we'll
+// try to stop someone swallowing the world in one gulp.
+const maxAlloc = 10000000
```

Comment:
I'm not really confident in having a limit like this. It's small enough that a large program could conceivably hit it (particularly if you want to allocate `[0] * N`, then fill in the values later). But it's large enough that minimal platforms could still OOM. And it doesn't protect from malicious inputs that OOM.

Reply:
The purpose of the limit is to prevent the most trivial mistakes from killing the program (and possibly the machine---sadly this happens all too often), not to stop malicious DoS, which is essentially impossible in a GC language.

I agree it's imperfect, but every memory allocator imposes some limit. The Go runtime caps malloc at 0x1000 << 32 on 64-bits or 1<<32 on a small machine, but in reality bad things happen long before you get there. I would rather start with a small limit and raise it as needed (probably never).